src/chain.h

```diff
@@ -512,7 +536,15 @@ class CDiskBlockIndex : public CBlockIndex
         } else if (ser_action.ForRead()) {
             // For block indices written before the client was Heartwood-aware,
             // these are always identical.
-            hashFinalSaplingRoot = hashLightClientRoot;
+            hashFinalSaplingRoot = hashBlockCommitments;
+        }
+
+        // Only read/write NU5 data if the client version used to create this
+        // index was storing them. For block indices written before the client
+        // was NU5-aware, these are always null / zero.
+        if ((s.GetType() & SER_DISK) && (nVersion >= NU5_DATA_VERSION)) {
+            READWRITE(hashAuthDataRoot);
+            READWRITE(hashFinalOrchardRoot);
         }
```

Comment:
Do we want to explicitly set these fields to null / zero rather than relying on the default constructor?

Reply:
De-serialization in this codebase always follows the pattern:
```
TypeName foo;
ss >> foo;
```
which ensures we start from a nulled type.